### PR TITLE
Add updateInterval args

### PR DESCRIPTION
**Pull request by devincd**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by hzxuzhonghu — APPROVED** [review]
lgtm

File: releasenotes/notes/add-updateinterval-args.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: traffic-management
+releaseNotes:
+  - |
+    **Added** a flag setting `updateInterval` in pilot and its default value is 500ms.
```
Comment (hzxuzhonghu) [inline releasenotes/notes/add-updateinterval-args.yaml:6]: It needs to be more clear about what it means

File: pilot/cmd/pilot-discovery/app/cmd.go [hunk 2]
```diff
@@ -137,6 +137,8 @@ func addFlags(c *cobra.Command) {
 		"comma separated list of networking plugins to enable")
 	c.PersistentFlags().DurationVar(&serverArgs.ShutdownDuration, "shutdownDuration", 10*time.Second,
 		"Duration the discovery server needs to terminate gracefully")
+	c.PersistentFlags().DurationVar(&serverArgs.StatusReporterOptions.UpdateInterval, "updateInterval", 500*time.Millisecond,
+		"Interval duration the reporter updates status")
```
Comment (hzxuzhonghu) [inline pilot/cmd/pilot-discovery/app/cmd.go:141]: `The interval that status reporter updates status`

**Comment by ramaraochavali** [comment]
Do we want to do this via command line arg or env var? Right now status is enabled via Env var and I guess at some point it might become part of meshconfig?
cc: @howardjohn

File: pilot/cmd/pilot-discovery/app/cmd.go [hunk 3]
```diff
@@ -137,6 +137,8 @@ func addFlags(c *cobra.Command) {
 		"comma separated list of networking plugins to enable")
 	c.PersistentFlags().DurationVar(&serverArgs.ShutdownDuration, "shutdownDuration", 10*time.Second,
 		"Duration the discovery server needs to terminate gracefully")
+	c.PersistentFlags().DurationVar(&serverArgs.StatusReporterOptions.UpdateInterval, "updateInterval", 500*time.Millisecond,
+		"The interval that status reporter updates status")
```
Comment (brian-avery) [inline pilot/cmd/pilot-discovery/app/cmd.go:141]: How does this sound?

> Interval to update the XDS distribution status.

File: releasenotes/notes/add-updateinterval-args.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: traffic-management
+releaseNotes:
+  - |
+    **Added** a flag setting `updateInterval` that is the interval that status reporter updates status and its default value is 500ms.
```
Comment (brian-avery) [inline releasenotes/notes/add-updateinterval-args.yaml:6]: Based on the above comment: 

```suggestion
    **Added** a flag, `updateInterval`, to configure the interval to update the XDS distribution status at. This defaults to 500ms.
```

**Comment by devincd** [comment]
> We have been trying to move away from using command line arguments which present a number of problems. Generally the progression is environment variable (experimental) -> mesh config (when not experimental).
> 
> Could you describe the use case for configuring this?

I just want to let `UpdateInterval` (the interval that status reporter updates status) to be configurable, instead of the default 500ms. So using command line arguments, environment variable and mesh config are both ok. I will submit another PR with environment variable.

File: pilot/cmd/pilot-discovery/app/cmd.go (hunk 3 above)
Comment (devincd) [reply pilot/cmd/pilot-discovery/app/cmd.go:141]: It sounds good. Thinks

**Pull request closed without merging**